File: 20220727/모범답안/20220727_01.js/CountEven_test.js
import assert from 'assert'
import solution from './CountEven.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const arr = [1, 9, 9, 1, 3, 2, 6]

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(2, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const arr = [1, 72, 24]

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(2, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const arr = [3, 3, 3, 3, 10]

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(1, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const arr = [8, 3, 6, 2, 6, 1, 5, 9, 7, 4]

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(5, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const arr = []

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(0, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const arr = [90, 7, 28, 69, 87, 98, 85, 16, 54, 18]

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(6, result)
    done();
  });
})

File: 20220727/모범답안/20220727_01.js/CountEven_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.supereasy.CountEven.CountEven import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        arr = [1, 9, 9, 1, 3, 2, 6]

        # when
        result = solution(arr)

        # then
        self.assertEqual(2, result)

    def test_2(self):
        # given
        arr = [1, 72, 24]

        # when
        result = solution(arr)

        # then
        self.assertEqual(2, result)

    def test_3(self):
        # given
        arr = [3, 3, 3, 3, 10]

        # when
        result = solution(arr)

        # then
        self.assertEqual(1, result)

    def test_4(self):
        # given
        arr = [8, 3, 6, 2, 6, 1, 5, 9, 7, 4]

        # when
        result = solution(arr)

        # then
        self.assertEqual(5, result)

    def test_5(self):
        # given
        arr = []

        # when
        result = solution(arr)

        # then
        self.assertEqual(0, result)

    def test_6(self):
        # given
        arr = [90, 7, 28, 69, 87, 98, 85, 16, 54, 18]

        # when
        result = solution(arr)

        # then
        self.assertEqual(6, result)


if __name__ == '__main__':
    unittest.main()
